src/test/java/org/padaiyal/utilities/vaidhiyar/JvmUtilityTest.java

```diff
@@ -464,4 +517,188 @@ void testGetAllocatedMemoryInBytesWithInvalidInput(long threadId) {
+   *
+   * @param memoryType Type of memory to get usage values and test.
+   */
+  @ParameterizedTest
+  @CsvSource({
+      "heap",
+      "nonHeap"
+  })
+  void testGetMemoryUsage(String memoryType) {
+    double memoryUsagePercentageLowerLimit = -1.0;
+    double memoryUsagePercentageUpperLimit = -1.0;
+    long maxMemory = Long.MAX_VALUE;
+    ExtendedMemoryUsage extendedMemoryUsage = null;
+    if (memoryType.equals("heap")) {
+      extendedMemoryUsage = JvmUtility.getHeapMemoryUsage();
+      memoryUsagePercentageLowerLimit = 0.0;
+      memoryUsagePercentageUpperLimit = 100.0;
+      maxMemory = extendedMemoryUsage.getMax();
+    } else if (memoryType.equals("nonHeap")) {
+      extendedMemoryUsage = JvmUtility.getNonHeapMemoryUsage();
+    }
+
+    Assertions.assertNotNull(
+        extendedMemoryUsage,
+        I18nUtility.getFormattedString(
+            "JvmUtilityTest.error.invalidValue",
+            memoryType,
+            "memoryType"
+        )
+    );
+
+    testNumericRange(
+        memoryType + " memoryUsagePercentage",
+        extendedMemoryUsage.getMemoryUsagePercentage(),
+        memoryUsagePercentageLowerLimit,
+        memoryUsagePercentageUpperLimit
+    );
+    testNumericRange(
+        memoryType + " memoryInit",
+        extendedMemoryUsage.getInit(),
+        0.0,
+        maxMemory
+    );
+    testNumericRange(
+        memoryType + " memoryUsed",
+        extendedMemoryUsage.getUsed(),
+        0.0,
+        extendedMemoryUsage.getCommitted()
+    );
+    testNumericRange(
+        memoryType + " memoryCommitted",
+        extendedMemoryUsage.getCommitted(),
+        extendedMemoryUsage.getUsed(),
+        maxMemory
+    );
+    testNumericRange(
+        memoryType + " memoryMax",
+        maxMemory,
+        extendedMemoryUsage.getCommitted(),
+        maxMemory
+    );
+  }
+
+  /**
+   * Test JvmUtility::getGarbageCollectionInfos().
+   */
+  @Test
+  void testGarbageCollectionInfo() {
+    GarbageCollectionInfo[] garbageCollectionInfos = JvmUtility.getGarbageCollectionInfo();
+
+    Assertions.assertNotNull(garbageCollectionInfos);
+
+    Arrays.stream(garbageCollectionInfos)
+        .peek(garbageCollectionInfo -> logger.info(
+            I18nUtility.getFormattedString(
+                "JvmUtilityTest.info.garbageCollectionInfo",
+                garbageCollectionInfo.getName(),
+                garbageCollectionInfo.getCollectionCount(),
+                garbageCollectionInfo.getCollectionTime().toMillis(),
+                "ms"
+            )
+          )
+        )
+        .forEach(garbageCollectionInfo -> {
+          Assertions.assertNotNull(garbageCollectionInfo.getName());
+          testNumericRange(
+              "GC collection count",
+              garbageCollectionInfo.getCollectionCount(),
+              0,
+              Long.MAX_VALUE
+          );
+          testNumericRange(
+              "GC collection time",
```

Comment:
doesn't this need to internationalized?

Reply:
Nope. It's referring to a field, so translating it will just add additional overhead.